share/rocm/cmake/ROCMCheckTargetIds.cmake

```diff
@@ -19,11 +19,23 @@ function(rocm_check_target_ids VARIABLE)
     endif()
 
     foreach(_target_id ${PARSE_TARGETS})
-        set(_result_var "HAVE_${_target_id}")
+        _rocm_sanitize_target_id("${_target_id}" _result_var)
+        set(_result_var "HAVE_${_result_var}")
         check_cxx_compiler_flag("-xhip --offload-arch=${_target_id}" "${_result_var}")
         if(${_result_var})
             list(APPEND _supported_target_ids "${_target_id}")
         endif()
     endforeach()
-    set(${VARIABLE} "${_supported_target_ids}" PARENT_SCOPE)
+    set("${VARIABLE}" "${_supported_target_ids}" PARENT_SCOPE)
+endfunction()
+
+function(_rocm_sanitize_target_id TARGET_ID VARIABLE)
```

Comment:
Can you link to a code snippet that will use this function? Do we need a test for this function in rocm-cmake?

Reply:
I mean a link to a problem you are facing in a problem outside of rocm-cmake. I want to understand more about the problem you are facing.

Reply:
I actually found this with rocPRIM, but it does not actually cause issues for us. Since I was curious I looked at why its failing and made a simple fix. Here's a simple repoducer: https://github.com/Maetveis/rocm-cmake-repro.

The second function is only for internal use in rocm-cmake thats why its name starts with an underscore. Regarding tests I didn't find any in the code for rocm_check_target_ids, but I can add one for this case. The only issue with that is the detection needs hipcc or hip-clang and currently it is not part of this repro's CI.

Reply:
Hi @lawruble13, should we add CI for rocm-cmake to a CI system that has hipcc available? No certain if it worth the extra effort but I feel more confident if all code paths are tested.

Reply:
We dont need hipcc just clang installed. We can setup dockers with github actions to do this.